src/ray/object_manager/test/pull_manager_test.cc

```diff
@@ -422,6 +440,58 @@ TEST_P(PullManagerTest, TestBasic) {
   AssertNoLeaks();
 }
 
+TEST_P(PullManagerTest, TestPinActiveObjects) {
+  auto prio = BundlePriority::TASK_ARGS;
+  if (GetParam()) {
+    prio = BundlePriority::GET_REQUEST;
+  }
+  auto refs = CreateObjectRefs(3);
+  auto oids = ObjectRefsToIds(refs);
+  AssertNumActiveRequestsEquals(0);
+  std::vector<rpc::ObjectReference> objects_to_locate;
+  auto req_id = pull_manager_.Pull(refs, prio, &objects_to_locate);
+  ASSERT_EQ(ObjectRefsToIds(objects_to_locate), oids);
+
+  std::unordered_set<NodeID> client_ids;
+  client_ids.insert(NodeID::FromRandom());
+  for (size_t i = 0; i < oids.size(); i++) {
+    ASSERT_FALSE(pull_manager_.IsObjectActive(oids[i]));
+    pull_manager_.OnLocationChange(oids[i], client_ids, "", NodeID::Nil(), 1);
+  }
+  for (size_t i = 0; i < oids.size(); i++) {
+    ASSERT_TRUE(pull_manager_.IsObjectActive(oids[i]));
+  }
+  ASSERT_EQ(num_send_pull_request_calls_, oids.size());
+  ASSERT_EQ(num_restore_spilled_object_calls_, 0);
+  AssertNumActiveRequestsEquals(oids.size());
+  pull_manager_.UpdatePullsBasedOnAvailableMemory(4);
+
+  // Check we pin objects belonging to active bundles.
```

Comment:
Can you also check:
- pinning objects that are already local when the request is activated
- do not pin objects that are part of bundles that are registered but not activated

It would also be nice to add a regression test that makes sure that `UpdatePullsBasedOnAvailableMemory` reflects the pinned objects size.

Reply:
The second is already tested below. I think the first is ok to skip-- we'd have to go out of our way to break that.

For the integration test, I think that would be best done after a larger object manager refactoring.

Reply:
For the second, it looks like the test only checks for objects that aren't registered at all. It would be nice to also check that objects that are inactive due to throttling do not get pinned.